flag/service/resource/resource.go

```diff
@@ -5,6 +5,7 @@ import (
 )
 
 type Resource struct {
-	ConfigMap configmap.ConfigMap
-	Retries   string
+	CertificateDirectory string
```

Comment:
This is here, as opposed to at a `ConfigMap` level, as it needs to be the same for both `configmap` and `certificate` resources.